### PR TITLE
feat(webpack-config): add mjs support

### PR DESCRIPTION
**Pull request by EvanBacon**
First step for adding support for `nomodule` will be to add support for `.mjs`.
Also added static path to `app.json` and unified across customize and webpack.

### CODE REVIEW

**Comment by ide** [comment]
Adding .mjs is fine I think but FYI browsers don't care about the extension and neither will Node -- the module system plans all work with .js.

**Comment by EvanBacon** [comment]
I'm more interested in ensuring that nomodule webpack plugins work correctly.

File: packages/webpack-config/webpack/webpack.common.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ module.exports = function(env = {}, argv) {
   const mode = getMode(env);
   const isDev = mode === 'development';
   const isProd = mode === 'production';
-
+  const useTypeScript = true;
```
Comment (ide) [inline packages/webpack-config/webpack/webpack.common.js]: ?

File: packages/webpack-config/webpack/utils/getPaths.js [hunk 2]
```diff
@@ -2,25 +2,44 @@ const path = require('path');
...
+    fs.existsSync(resolveFn(`${filePath}.${extension}`))
+  );
+
+  if (extension) {
+    return resolveFn(`${filePath}.${extension}`);
+  }
+
+  return null;
+};
+const resolveModule = (resolveFn, filePath) => {
```
Comment (ide) [inline packages/webpack-config/webpack/utils/getPaths.js]: remove?

File: packages/webpack-config/webpack/utils/getPaths.js [hunk 3]
```diff
@@ -2,25 +2,44 @@ const path = require('path');
...
+  'web.tsx',
+  'tsx',
+  'json',
+  'web.jsx',
+  'jsx',
 ];
 
+const resolveModuleExists = (resolveFn, filePath) => {
+  const extension = extensions.find(extension =>
+    fs.existsSync(resolveFn(`${filePath}.${extension}`))
```
Comment (ide) [inline packages/webpack-config/webpack/utils/getPaths.js]: sometimes people already include extensions in their module names e.g. `require('./x.js')`

File: packages/webpack-config/webpack/webpack.common.js (hunk 1 above)
Comment (EvanBacon) [reply packages/webpack-config/webpack/webpack.common.js]: future plans. Keeping this PR simple

**Comment by EvanBacon** [comment]
#1075

**Pull request closed without merging**